geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/IndexRepositoryFactory.java

```diff
@@ -102,6 +105,19 @@ private IndexRepository finishComputingRepository(Integer bucketId, LuceneSerial
     if (oldRepository != null) {
       oldRepository.cleanup();
     }
+
+    boolean hasOldMember = false;
+    if (userRegion.getCache() != null) {
```

Comment:
We can get this as a cache helper method to prevent repeated code. We can make it a part of the cache.
We can implement it how cache.hasPersistenceRegion() is implemented.
boolean cache.hasMemberOlderThan(Version version){<implementation>}